sfdoc/publish/tests/utils.py

```diff
@@ -41,116 +50,145 @@ def create_test_html(url_name, title, summary, body):
 
 
 def mock_update_draft(instance_url, kav_id):
-    url = urljoin(instance_url, 'services/data/v{}/sobjects/{}/{}'.format(
-        settings.SALESFORCE_API_VERSION,
-        settings.SALESFORCE_ARTICLE_TYPE,
-        kav_id,
-    ))
-    responses.add(
-        'PATCH',
-        url=url,
-        status=HTTPStatus.NO_CONTENT,
+    url = urljoin(
+        instance_url,
+        "services/data/v{}/sobjects/{}/{}".format(
+            settings.SALESFORCE_API_VERSION, settings.SALESFORCE_ARTICLE_TYPE, kav_id
+        ),
+    )
+    responses.add("PATCH", url=url, status=HTTPStatus.NO_CONTENT)
+
+
+ORIGINIT = TemporaryDirectory.__init__
+
+
+@contextmanager
+def makeDebugTemporaryDirectoryMock(parent_prefix="", default_dir=""):
+    class DebugTemporaryDirectory(TemporaryDirectory):
+        def __init__(self, suffix=None, prefix="", dir=None):
+            full_prefix = parent_prefix
+            full_prefix += "_" + str(time.process_time()) + "_"
+            if getattr(self, "subprefix", None):
+                full_prefix += "_" + self.subprefix
+            if prefix:
+                full_prefix += "_" + prefix
+            ORIGINIT(self, prefix=full_prefix, dir=(dir or default_dir))
+
+        @classmethod
+        def set_subprefix(cls, subprefix):
+            print("CLASS", cls)
+            cls.subprefix = subprefix
+
+        def cleanup(self, name="", warn_message=""):
+            pass
+
+    d = DebugTemporaryDirectory
+    with mock.patch("tempfile.TemporaryDirectory.__init__", d.__init__),\
+        mock.patch("tempfile.TemporaryDirectory.cleanup", d.cleanup),\
+        mock.patch("tempfile.TemporaryDirectory._cleanup", d.cleanup): # noqa E125
+            TemporaryDirectory.set_subprefix = classmethod(d.set_subprefix.__func__) # noqa E117
+            yield TemporaryDirectory
+
+
+def integration_mocks():
+    for filename in glob.glob(os.path.join(rootdir, "testdata/bundles/*.zip")):
+        UUID = os.path.splitext(os.path.basename(filename))[0]
+        url = f"https://salesforce.easydita.com/rest/all-files/{UUID}/bundle"
+        responses.add(responses.GET, url, body=open(filename, "rb"))
+    responses.add_passthru("https://test.salesforce.com")
+
+    def pass_thru(request):
+        response = responses._real_send(responses.HTTPAdapter(), request)
+        return (response.status_code, response.headers, response.raw.data)
+
+    responses.add_callback(
+        method=responses.GET,
+        url=re.compile("https://.*.salesforce.com/.*"),
+        callback=pass_thru,
+    )
+    responses.add_callback(
+        method=responses.POST,
+        url=re.compile("https://.*.salesforce.com/.*"),
+        callback=pass_thru,
+    )
+    responses.add_callback(
+        method=responses.PATCH,
+        url=re.compile("https://.*.salesforce.com/.*"),
+        callback=pass_thru,
+    )
+    responses.add_callback(
+        method=responses.GET,
+        url=re.compile("https://.*.salesforce.com/.*"),
+        callback=pass_thru,
```

Comment:
This one is the same as on line 175

Reply:
For the record, this is all of the stuff I had to add because responses doesn't regexps in this particular context (even though it does in others). This is what I wanted to write:
 
    responses.add_passthru("https://*.salesforce.com/*")

But I had to write ~20 lines of code instead.

Reply:
94eb3bab530079e788a9d30b4bd4f7f881ec00ed